api/v1/service/resource/resource.go

```diff
@@ -33,6 +34,8 @@ type service struct {
 
 var replacerStrings = []string{"github.com", "raw.githubusercontent.com", "/tree/", "/", "/blob/", "/raw/", "/src/", "/raw/"}
 
+const catalogCloneBasePath = "/tmp/catalog"
```

Comment:
you can write a function in the pkg which clones 
and it returns this path
instead of defining it again here